File: src/core/models/member.ts
import { User } from "./user.ts";

export class Member extends User{
    readonly dateOfBirth: Date;
    readonly age: number;
    readonly gender: string;

    constructor(
        userUUID: string,
        userCode: string,
        firstname: string,
        surname: string,
        lastname: string,
        email: string,
        roles: string[],
        createdDate: Date,
        dateOfBirth: Date,
        age: number,
        gender: string
    ) {
        super(userUUID, userCode, firstname, surname, lastname, email, createdDate);
        this.dateOfBirth = dateOfBirth;
        this.age = age;
        this.gender = gender;
    }
}


export class MemberInfo {
    readonly userUUID: string;
    readonly dateOfBirth: Date;
    readonly age: number;
    readonly gender: string;

    constructor(
        userUUID: string,
        dateOfBirth: Date,
        age: number,
        gender: string
    ) {
        this.userUUID = userUUID;
        this.dateOfBirth = dateOfBirth;
        this.age = age;
        this.gender = gender;
    }
}

File: lib/src/core/models/member.dart

import 'package:salud_ulv_app/src/core/models/user.dart';

class Member extends User{
  // final String? groupID;
  final DateTime? dateOfBirth;
  final int? age;
  //final int tier
  final String gender;
  // final String? occupation;

  Member({
    super.userUUID,
    super.userCode,
    required super.firstname,
    required super.surname,
    required super.lastname,
    required super.email,
    super.currentRole,
    super.roles,
    super.createdAt,

    this.dateOfBirth,
    required this.gender,
    this.age,

    // this.groupID,
    // this.occupation,
  });
  
}


class MemberInfo {
  String? userUUID;
  final int? groupID;
  final int? typeID;
  final DateTime? dateOfBirth;
  int? age;
  final String? gender;

  MemberInfo({
    this.userUUID,
    this.groupID,
    this.typeID,
    this.dateOfBirth,
    this.age,
    this.gender,
    // required this.occupation,
  });
}
